addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5331,6 +5417,80 @@ registry.layout_column = SnippetOptionWidget.extend({
         rowEl.classList.remove('o_we_padding_highlight');
         delete rowEl._removePaddingPreview;
     },
+    /**
+     * Checks if the current device is mobile.
+     *
+     * @private
+     * @returns {boolean}
+     */
+    _isMobile() {
+        let isMobile;
+        this.trigger_up('service_context_get', {
+            callback: (ctx) => {
+                isMobile = ctx['isMobile'];
+            },
+        });
+
+        return isMobile;
+    },
+    /**
+     * Calculates the number of columns depending on the mobile or desktop version.
+     *
+     * @private
+     * @param {boolean} isMobile
+     * @returns {integer} number of columns depending on mobile/desktop
+     */
+    _getNbColumns(isMobile) {
+        const items = this.$target[0].querySelector(':scope > .row')?.children;
+        if (isMobile && items) {
+            const itemSize = this._getItemSize(items[0], isMobile);
+            if (itemSize * items.length > 12 ) {
+                const nbColumns = 12 / itemSize;
+                return nbColumns;
+            }
+        }
+        return items?.length || 0;
+    },
+    /**
+     * Gets the current size of an item, depending on whether it is on mobile or desktop.
+     *
+     * @private
+     * @param {Element} item
+     * @param {boolean} isMobile
+     */
+    _getItemSize(item, isMobile) {
+        const resolutionModifier = isMobile ? '' : '-lg';
+        const resolutionRegex = new RegExp(`\\bcol${resolutionModifier}(-(\\d{1,2}))?(?![\\w-])`);
+        const columnClass = item.className.match(resolutionRegex);
+        return columnClass ? columnClass[2] : 12;
+    },
+    /**
+     * Calculates the offset to apply to an item.
+     *
+     * @private
+     * @param {integer} itemsOnLastRow - number of items on the last row
+     * @param {integer} itemSize
+     * @param {integer} nbColumns
+     * @returns {integer}
+     */
+    _getItemOffset(itemsOnLastRow, itemSize, nbColumns) {
+        return itemsOnLastRow ? Math.floor((12 - itemSize * itemsOnLastRow) / 2) : Math.floor((12 - itemSize * nbColumns) / 2);
+    },
+    /**
+     * Adds the offset when displayed on mobile to the proper item.
+     *
+     * @private
+     * @param {(jQuery|HTMLCollection)} $items
+     * @param {integer} itemsOnLastRow - number of items on the last row
+     * @param {integer} itemOffset - offset to add to the item
+     */
+    _addMobileOffset($items, itemsOnLastRow, itemOffset) {
+        const targetItem = $items[$items.length - itemsOnLastRow];
```

Comment:
As `$items[$items.length - itemsOnLastRow]` is a HTML node, the variable name should finish by `El`